src/tools/pbs_upgrade_job.c

```diff
@@ -194,224 +286,258 @@ check_job_file(int fd)
+				"%s", old_jobfix_pre19.ji_queue);
+		snprintf(new_job.ji_qs.ji_destin, sizeof(new_job.ji_qs.ji_destin),
+				"%s", old_jobfix_pre19.ji_destin);
+		new_job.ji_qs.ji_un_type = old_jobfix_pre19.ji_un_type;
+		memcpy(&new_job.ji_qs.ji_un, &old_jobfix_pre19.ji_un, sizeof(new_job.ji_qs.ji_un));
+	}
+		/* Open a temporary file to stage data */
+		tmp = tmpfile();
+		if (!tmp) {
+			fprintf(stderr, "Failed to open temporary file [%s]\n",
+					errno ? strerror(errno) : "No error");
+			return 1;
+		}
+		tmpfd = fileno(tmp);
+		if (tmpfd < 0) {
+			fprintf(stderr, "Failed to find temporary file descriptor [%s]\n",
+					errno ? strerror(errno) : "No error");
+			return 1;
+		}
 
+		/* Write the new jobfix structure to the output file */
+		len = write(tmpfd, &new_job.ji_qs, sizeof(new_job.ji_qs));
+		if (len != sizeof(new_job.ji_qs)) {
+			fprintf(stderr, "Failed to write jobfix to output file [%s]\n",
+					errno ? strerror(errno) : "No error");
+			return 1;
+		}
+
+		/* Read the rest of the input and write it to the temporary file */
+		do {
+			len = read(fd, buf, sizeof(buf));
+			if (len < 0) {
+				fprintf(stderr, "Failed to read input file [%s]\n",
+						errno ? strerror(errno) : "No error");
+				return 1;
+			}
+			if (len < 1)
+				break;
+			len = write(tmpfd, buf, len);
+			if (len < 0) {
+				fprintf(stderr, "Failed to write output file [%s]\n",
+						errno ? strerror(errno) : "No error");
+				return 1;
+			}
+		} while (len > 0);
+
+		/* Reset the file descriptors to zero */
+		pos = lseek(fd, 0, SEEK_SET);
+		if (pos != 0) {
+			fprintf(stderr, "Failed to reset job file position [%s]\n",
+					errno ? strerror(errno) : "No error");
+			return 1;
+		}
+		pos = lseek(tmpfd, 0, SEEK_SET);
+		if (pos != 0) {
+			fprintf(stderr, "Failed to reset temporary file position [%s]\n",
+					errno ? strerror(errno) : "No error");
+			return 1;
+		}
+
+		/* Copy the data from the temporary file back to the original */
+		do {
+			len = read(tmpfd, buf, sizeof(buf));
+			if (len < 0) {
+				fprintf(stderr, "Failed to read temporary file [%s]\n",
+						errno ? strerror(errno) : "No error");
+				return 1;
+			}
+			if (len < 1)
+				break;
+			len = write(fd, buf, len);
+			if (len < 0) {
+				fprintf(stderr, "Failed to write job file [%s]\n",
+						errno ? strerror(errno) : "No error");
+				return 1;
+			}
+		} while (len > 0);
+
+		ret = fclose(tmp);
+		if (ret != 0) {
+			fprintf(stderr, "Failed to close temporary file [%s]\n",
+					errno ? strerror(errno) : "No error");
+			return 1;
+		}
 	return 0;
 }
 
 /**
  * @brief
  *		Upgrade a task file from an earlier version.
  *
- * @param[in]	taskfile	-	File name of the task file
- *
+ * @param[in]	taskfile		-	File name of the task file
+ * @param[in]	taskfile_version	-	Version of the job taskfix structure 
  * @return	int
  * @retval	-1	: failure
  * @retval	0	: success
  */
 int
-upgrade_task_file(char *taskfile)
+upgrade_task_file(char *taskfile, int taskfile_version)
```

Comment:
Here also no need to pass taskfile_version if we clean-up the code for older versions of PBS and then this function gets called only for pre-19 case then.

Reply:
done!